Microsoft.Toolkit.Uwp.UI.Controls/TokenizingTextBox/InterspersedObservableCollection.cs

```diff
@@ -193,8 +195,15 @@ private void ReadjustKeys()
         private int ToInnerIndex(int outerIndex)
         {
 #if DEBUG
-            Guard.IsInRange(outerIndex, 0, Count, nameof(outerIndex));
-            Guard.IsFalse(_interspersedObjects.ContainsKey(outerIndex), nameof(outerIndex)); // We can't map a inserted key to the original collection!
+            if ((uint)outerIndex >= Count)
```

Comment:
Do we have to worry about the lower-bound check here that was originally done by `IsInRange` or is that handled by the (uint) conversion?

Reply:
The lower bound is handled by the `(uint)` cast, it's the same trick I use in the `Guard` APIs as well to skip one conditional branch and checks both edges of the range with just one. This works because signed integers are stored in 2-complemement, with the most significant bit being the sign. If you cast to `uint` instead, if the value was negative, it'd just underflow and goes to `int.MaxValue + x - 1`, which will always be greater than the other comparand since that's guaranteed to be a non-negative value in the `int` range 🙂